sources/packages/backend/libs/services/src/notifications/notification/notification.service.ts

```diff
@@ -120,6 +121,34 @@ export class NotificationService extends RecordDataModelService<Notification> {
     return await this.processUnsentNotificationsRecursive(pollingRecordsLimit);
   }
 
+  /**
+   * Checks if a notification exists for the given optional parameters provided.
+   * @param transactionalEntityManager entity manager to be part of the transaction.
+   * @param options related options.
+   * - `applicationNumber` related application number.
+   * @returns boolean true if the notification exists, else false.
+   */
+  async checkNotificationExists(
+    transactionalEntityManager: EntityManager,
+    options?: { applicationNumber?: string },
```

Comment:
This does not seem optional for the method.
Either this method is generic or dedicated to `ApplicationEditedTooManyTimesNotification` and must be renamed.
At this moment do whatever would be faster.